Zend/Optimizer/zend_dfg.c

```diff
@@ -25,9 +25,11 @@ static zend_always_inline void _zend_dfg_add_use_def_op(const zend_op_array *op_
 	const zend_op *next;
 
 	if (opline->op1_type & (IS_CV|IS_VAR|IS_TMP_VAR)) {
-		var_num = EX_VAR_TO_NUM(opline->op1.var);
-		if (!zend_bitset_in(def, var_num)) {
-			zend_bitset_incl(use, var_num);
+		if (!(build_flags & ZEND_DFG_SHORT_CLOSURE) || opline->opcode != ZEND_ASSIGN) {
+			var_num = EX_VAR_TO_NUM(opline->op1.var);
+			if (!zend_bitset_in(def, var_num)) {
+				zend_bitset_incl(use, var_num);
+			}
```

Comment:
zend_build_dfg produces a use for every variable, including the lhs of assigns, which was not desirable (in `$a = 1;`, `$a` appeared in live variables when it was never readden before that).

Other ASSIGN ops (e.g. ASSIGN_DIM) need to read their lhs, so they are not included here.

I need to dig in other ops.